src/searchlog.c

```diff
@@ -373,37 +375,39 @@ int bandstr2line(char *buffer) {
 //
 // return true if the qso was in the same mode as the current one
 //
-static bool is_current_mode(const char *qso) {
+static bool qso_has_current_mode(const struct qso_t *qso) {
 
     if (!mixedmode) {
 	return true;    // always true if not in mixed mode
     }
 
-    return log_get_mode(qso) == trxmode;
+    return qso->mode == trxmode;
 }
 
 
-/* search complete Log for 'hiscall' as substring in callsign field and
+/* search complete Log for 'call' as substring in callsign field and
  * copy found QSO to 'searchresults'. Extract relevant data to 'result'.
- * */
-void filterLog() {
-
-    char s_inputbuffer[LOGLINELEN + 1] = "";
+ */
+void filterLog(const char *call) {
 
     srch_index = 0;
 
-    for (int qso_index = 0; strlen(qsos[qso_index]) > 4; qso_index++) {
+    for (int qso_index = 0; qso_index < nr_qsos; qso_index++) {
+
+	struct qso_t *qso = g_ptr_array_index(qso_array, qso_index);
+	if (qso->is_comment) {
+	    continue;
+	}
 
-	if (!is_current_mode(qsos[qso_index])) {
-	    continue;   // different mode
+	if (!qso_has_current_mode(qso)) {
+	    continue;	// different mode
 	}
 
-	g_strlcpy(s_inputbuffer, qsos[qso_index] + 29, 13); /* call */
-	if (strstr(s_inputbuffer, hiscall) == 0) {
+	if (strstr(qso->call, call) == 0) {
 	    continue;   // no match
 	}
 
-	g_strlcpy(searchresult[srch_index], qsos[qso_index], 81);
+	g_strlcpy(searchresult[srch_index], QSOS(qso_index), 81);
```

Comment:
Just for the TODO list: rework searching to use parsed QSO data. Now it practically parses log line again in `extractData()`.